src/app/campaign.service.spec.ts

```diff
@@ -110,17 +110,17 @@ describe('CampaignService', () => {
     expect(CampaignService.isInFuture(campaign)).toBe(false);
   });
 
-  it ('should allow donation attempts to any campaign in active date range, even if Status gets stuck in Preview', () => {
+  it ('should not allow donation attempts to campaigns in Preview status', () => {
```

Comment:
@tbg-jasonfung this seems problematic because the whole point of this test & logic was to make sure we weren't relying solely on the Status, which had a delay updating. By reverting that we will reinstate the bug we had before.

Reply:
@NoelLH Yeah I thought that too, but I think that was when we were using `Campaign Status` rather than the refactored `Status` which is a much more reactive formula field - i've been testing flicking the status between `Preview` and `Active` and frontend is reacting immediately

Reply:
Alternatively, we could handle regular campaigns that have not had the appropriate approvals with a separate status like `Pending Approval`

Reply:
@tbg-jasonfung yeah – I woke up thinking about this! :/ And I don't think this can ever be reliable as things stand because we have to cache campaign data for up to a minute. So there is no way this can reliably flip for everyone at the right second. I definitely think a status that accurately captures these 2 different states is necessary, and that will probably mean no front end logic change except maybe listening for a different status. What we are currently calling _Pending_ is probably less confusing as something like _Scheduled_. So we would then have the understanding that Scheduled campaigns are about to launch if nothing changes, whereas Preview ones need some additional work, whether that's approval by us or by the charity or completion of required account info.

If it's simpler we could repurpose _Draft_, but I think the distinction between the reasons it's pending is probably useful and relied upon within Salesforce code too.